### DISCUSSION TITLE
Providing OpenTelemetry.Proto package

### DISCUSSION THREAD

**Discussion by mlavrent** [open-telemetry/opentelemetry-dotnet#5794] [Ideas]
Currently, the OpenTelemetry protobuf definitions are provided [here](https://github.com/open-telemetry/opentelemetry-proto).

Other languages provide bindings for these protobuf models as a package in their respective packaging systems:
- [Python](https://pypi.org/project/opentelemetry-proto/)
- [Go](https://github.com/open-telemetry/opentelemetry-proto-go)
- [Java](https://github.com/open-telemetry/opentelemetry-proto-java)
- [Rust](https://crates.io/crates/opentelemetry-proto/0.1.0)

This may be useful for folks wishing to write their own custom OTLP receiver or otherwise serialize/deserialize using the protobuf definitions. We could also factor the serialization/deserialization logic and protobuf files out from the [OTLP exporter](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol).

Propose to call this package `OpenTelemetry.Proto`, and essentially add [opentelemetry-proto](https://github.com/open-telemetry/opentelemetry-proto) as a submodule, and migrate the serialization logic from the OTLP package to this.

Want to hear your thoughts on if this is worth doing - if so, I'm happy to take this on to support some things I'm looking to do with this package as a standalone (as opposed to using it as part of the OTLP exporter).

**Comment by vishweshbankwar** [comment]
@mlavrent - This is not something we are planning to add/support. For .NET the plan to implement a custom protobuf serializer. 
https://github.com/open-telemetry/opentelemetry-dotnet/issues/5730.
